midcli/__main__.py

```diff
@@ -41,13 +42,33 @@ def __init__(self, websocket=None, user=None, password=None, command=None, inter
             editor = NonInteractiveEditor()
 
         self.context = Context(self, websocket=websocket, user=user, password=password,
-                               editor=editor, mode=mode)
+                               editor=editor, menu=menu, mode=mode)
         self.command = command
         self.completer = MidCompleter(self.context)
 
         self.last_kernel_message = None
         self.loop = None
 
+    def _build_menu(self):
+        def get_message():
+            prompt = '\n'.join([
+                f'{i + 1}) {title}'
+                for i, (title, _) in enumerate(menu_items)
```

Comment:
Just a small nit but `enumerate` takes a `start=` keyword that gives the starting position of the integer so you could do `start=1`.